packages/nuxt/src/core/runtime/nitro/renderer.ts

```diff
@@ -497,15 +497,15 @@ function renderHTMLDocument (html: NuxtRenderHTMLContext) {
 
 async function renderInlineStyles (usedModules: Set<string> | string[]): Promise<Style[]> {
   const styleMap = await getSSRStyles()
-  const inlinedStyles = new Set<string>()
```

Comment:
So with this setup duplicates would only happen if the user is writing/importing the same CSS across modules?

Reply:
It depends on how rollup chunks them, but the same CSS could be added if multiple components have the same chunk of CSS.

Looking at this with fresh eyes, I think probably we can't dedupe with module key _anyway_ as it's the content of the CSS that matters.